### ISSUE TITLE
The act of resetting a password should automatically mark a user as "verified"

### ISSUE THREAD

**Issue by burggraf** [pocketbase/pocketbase#4066]
Scenario:  The user signs up and never receives a confirmation email.  The user clicks on "forgot password" and gets a password reset link, click it, and changes their password.  At this point they are still "unverified", though I would argue that ARE verified since they received an email at the given address and clicked on it to verify the address is valid.

This would be the solution to "I didn't get a verification email" -- answer: "just reset your password."

**Comment by ganigeorgiev** [comment]
It makes sense, I'll consider it for the next release.

**Comment by burggraf** [comment]
I figured I could hack it with:
`onRecordAfterConfirmPasswordResetRequest`
but that doesn't seem to be firing when I click to reset my password, or maybe I'm misunderstanding that hook?

**Comment by ganigeorgiev** [comment]
The `onRecordAfterConfirmPasswordResetRequest` will be fired **after** the user clicks on the email link and successfully resets their password.

To avoid resaving the record, you can use its before hook, something like:

```js
onRecordBeforeConfirmPasswordResetRequest((e) => {
    if (e.record) {
        e.record.setVerified(true);
    }
})
```

**Comment by burggraf** [comment]
I just never see that trigger firing for some reason.

**Comment by burggraf** [comment]
Never mind, my mistake.  Environment isn't set up right.

**Comment by burggraf** [comment]
Yep, that works.  Hooks rock.

**Comment by ganigeorgiev** [comment]
This is implemented in the `develop` branch and will be available with the next v0.21.0 release (_there are no ETAs yet_).
